draco/analysis/flagging.py

```diff
@@ -627,6 +627,63 @@ def process(self, data):
         return data
 
 
+class NegativeAutosMask(task.SingleTask):
+    """Flag in frequency-time if any autocorrelation is negative."""
+
+    def setup(self, telescope):
+        """Save the telescope model.
+
+        Parameters
+        ----------
+        telescope : TransitTelescope
+            Telescope object to use
+        """
+        self.telescope = io.get_telescope(telescope)
+
+    def process(self, data: containers.VisContainer) -> containers.RFIMask:
+        """Extract autos and flag if any auto is negative.
+
+        Parameters
+        ----------
+        data
+            Timestream dataset containing visibilities and
+            the relevant index map entries.
+
+        Returns
+        -------
+        mask
+            time-frequency mask
+        """
+        data.redistribute("freq")
+
+        # Determine stack axis
```

Comment:
I don’t think `redefine_stack_index_map` changes which elements of the stack axis are autocorrelations.  It just updates which feeds are used to represent the "typical" autocorrelations, making sure they correspond to good feeds with valid positions, polarisations, etc. Since we’re not using any of the feed info here, I think we could skip this step and just use `data.prodstack` directly. That would also mean we don’t need to provide a telescope instance at setup. Let me know if I’m overlooking something!